index.d.ts

```diff
@@ -1,8 +1,8 @@
 import i18next, { ReactOptions, i18n, ThirdPartyModule, WithT, TFunction, Resource } from 'i18next';
 import * as React from 'react';
 
-type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
-type Subtract<T extends K, K> = Omit<T, keyof K>;
+export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
+export type Subtract<T extends K, K> = Omit<T, keyof K>;
```

Comment:
We don't want to expose internal types

Reply:
Ok. I only changed this because the TSLint rule "strict-export-declare-modifiers" noted it. Should I then deactivate this rule?
```
ts4.1/index.d.ts:194:6 - All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
```

Reply:
Small note:
You do not lint files with the `test:typescript` scripts. Correct would be for example
````
"test:typescript": "tslint --project tsconfig.json 'ts4.1/**/*.ts'",
````
instead of
```
"test:typescript": "tslint --project tsconfig.json",
```